src/main/fc/rc_controls.c

```diff
@@ -228,6 +233,14 @@ void processRcStickPositions()
     }
     doNotRepeat = true;
 
+    
+#if defined(USE_USB_CDC_HID) && defined(USE_VCP)
+    // If the FC is used as a joystick, stick commands should be inactive
+    if(usbVcpIsConnected() && usbDevConfig()->type == COMPOSITE) {
```

Comment:
Good find!

I think it makes sense to bring this check in line with the one that is done for CMS by checking if USB is connected **and** no battery is connected: https://github.com/betaflight/betaflight/blob/d3ad63a50bbb48c39a1a4d4d6ceaff4a95c9fba4/src/main/cms/cms.c#L1204-L1206

Ideally this can be done by refactoring the check into a `cdcDeviceMayBeActive()` (or suchlike) function.

Reply:
> Good find!

Thank you for the review @mikeller.
 
> I think it makes sense to bring this check in line with the one that is done for CMS by checking if USB is connected **and** no battery is connected:
> 
> https://github.com/betaflight/betaflight/blob/d3ad63a50bbb48c39a1a4d4d6ceaff4a95c9fba4/src/main/cms/cms.c#L1204-L1206
> 

The `(getBatteryCellCount() == 0 && usbDevConfig()->type == COMPOSITE)` already was the initial idea but... Some old boards (eg. Sparky2) have no VBAT pad and they have to be powered by external 5V BEC. When a V/C sensor is not connected, FC does not 'know' if it is connected via USB port or it's powered by the BEC. In case when FC is powered by the BEC the `(getBatteryCellCount() == 0 && usbDevConfig()->type == COMPOSITE)` will brick the rest of the method because its result will always be TRUE and stick commands in that case would never work unless the `usb_hid_cdc` is `OFF`. And that's the reason why the `usbVcpIsConnected()` was invoked here. I thought that method is a certain way for the FC to figure out if it's **really** connected via USB regardless of the BEC. However, if you think that using of the `usbVcpIsConnected()` is not appropriate here from some reason, I will accept your suggestion and use the  `getBatteryCellCount()` instead, no probs.

> Ideally this can be done by refactoring the check into a `cdcDeviceMayBeActive()` (or suchlike) function.

Yes, there is no reason for having a repeating code.
Where is the best place to write this conditional check? The code from the `cms.c` and the `rc_controls.c` should be able to access it.

Reply:
Correct. Looks like the check in `cms.c` is not ideal anyway, and should rather check for `getBatteryState() == BATTERY_NOT_PRESENT` than for `getBatteryCellCount()`, and combine this with a check to ensure that there is a voltage meter: `batteryConfig()->voltageMeterSource != VOLTAGE_METER_NONE && getBatteryState() == BATTERY_NOT_PRESENT`
I'd stick the shared code into `io/usb_cdc_hid.c` to keep it with other generic USB HID code.

Reply:
Ok, thanks.
I'll try on that way following your suggestion.
Can I do refactoring of `cms.c` in this PR too?

Reply:
Yes, I think this makes sense as part of this.